sequencer.go

```diff
@@ -1,42 +1,96 @@
 package desync
 
+import "sort"
+
 // SeedSequencer is used to find sequences of chunks from seed files when assembling
 // a file from an index. Using seeds reduces the need to download and decompress chunks
 // from chunk stores. It also enables the use of reflinking/cloning of sections of
 // files from a seed file where supported to reduce disk usage.
 type SeedSequencer struct {
-	seeds   []Seed
-	index   Index
-	current int
+	seeds             []Seed
+	index             Index
+	invalidSeedAction InvalidSeedAction
+	current           int
+}
+
+type matchingCandidate struct {
+	seed    Seed
+	source  SeedSegment
+	advance int
+}
+
+type matchingCandidateList []matchingCandidate
+
+func (m matchingCandidateList) Len() int {
+	return len(m)
+}
+
+func (m matchingCandidateList) Less(i, j int) bool {
+	// Sort the candidates in descending order based on the segment size
+	return m[i].source.Size() > m[j].source.Size()
+}
+
+func (m matchingCandidateList) Swap(i, j int) {
+	m[i], m[j] = m[j], m[i]
```

Comment:
The sort-interface here actually opens up more possibilities. What desync should have done from the beginning is prefer reflink over simple copies when dealing with seeds. It's much more efficient (time and space) to reflink a chain of lets say 3 chunks, than it is to copy 5 chunks, even though that's larger.

You don't need to do that now, but it's something that could be added on top of your change. If you are on a filesystem that supports reflinks that may be something to consider.